doc/user_manual/optimizer.tex

```diff
@@ -430,3 +430,286 @@ \subsubsection{Simultaneous Perturbation Stochastic Approximation (SPSA)}
+  in a ``relative'' sense: when the relative change of the objective variable in two successive model evaluations is smaller than
+  this specified threshold, the \xmlNode{FiniteDifferenceGradientOptimizer} optimizer is in convergence and terminates the simulation.
+      \default{1e-3}
+  \item \xmlNode{absoluteThreshold}, \xmlDesc{float, optional field}, specifies the convergence criteria to determine the optimality,
+  in an ``absolute'' sense: when the absolute change of objective variable in two successive model evaluations is smaller
+  than this specified threshold, the \xmlNode{FiniteDifferenceGradientOptimizer} optimizer is in convergence and terminates the simulation.
+      \default{0.0}
+  \item \xmlNode{gradientThreshold}, \xmlDesc{float, optional field}, specifies the convergence criteria to determine the optimality,
+   as function of the L2 norm of the gradient (useful for unconstrained problems): when the L2 norm od the gradient falls below this threshold, the \xmlNode{FiniteDifferenceGradientOptimizer} optimizer is in convergence and terminates the simulation.
+      \default{1e-3}
+  \item \xmlNode{minStepSize}, \xmlDesc{float, optional field}, specifies the minimum allowable step size in
+    the normalized input space, ranging from 0 (no movement) to 1 (spans any dimension).
+      \default{1e-9}
+  \item \xmlNode{gainGrowthSize}, \xmlDesc{float, optional field}, specifies the rate at which the step size
+    should grow when it does grow, for instance when multiple steps are in the same direction.  Increasing
+    this will increase the likelihood that an optimization path travels quickly across the domain along a
+    consistent gradient.
+      \default{2}
+  \item \xmlNode{gainShrinkSize}, \xmlDesc{float, optional field}, specifies the rate at which the step size
+    should shrink when it does shrink, for instance when switching directions on successive steps.  Increasing
+    this will slow convergence, but decrease the likelihood of achieving false convergence due to small step
+    sizes.
+      \default{same value as grainGrowthSize}
+  \end{itemize}
+\item \xmlNode{parameter}, \xmlDesc{XML node, optional parameter} will accepts the following sub-nodes:
+  \begin{itemize}
+  \item \xmlNode{numGradAvgIterations}, \xmlDesc{integer, optional field} is the number of iterations for gradient estimation. When this
+        parameter is $>1$, multiple gradient evaluations are going to be performed. Since the main goal for this parameter is to
+        get a better gradient estimation (performing a denoising), the current point $x_k$ is evaluated multiple times in order to be able to
+        converge in average.
+        \default{1}
+
+  \item Optimizer Gradient Evaluation parameters:
+    \begin{itemize}
+      \item \xmlNode{gamma}, \xmlDesc{float, optional field} Inverse exponent for gradient evaluation distance. Increasing this
```

Comment:
Similar, are these applicable/sensible for FE?

Reply:
We use the same machinery of the spsa for the FD optimizer. Those parameters are able to control the perturbation size in a reasonable way for now, so I think we should keep this.

Anyhow, as soon as this PR goes in, I will open an issue to remove all the redundancy on the documentation (here I just copied/pasted the one for SPSA removing the things not applicable to the FD)